BUG: scipy/io/arff/tests/test_arffread.py::TestNoData::test_nodata failure on big endian platforms

### Describe your issue.

When running the test suite on ppc64 (big endian), the following test fails:

```
FAILED scipy/io/arff/tests/test_arffread.py::TestNoData::test_nodata - AssertionError:
```

However, I don't really know if the test is expecting the wrong endianness, or if `loadarff()` is returning the type with wrong endianness. I suspect the latter.

### Reproducing Code Example

```python
n/a
```


### Error message

```shell
=============================================== FAILURES ===============================================
________________________________________ TestNoData.test_nodata ________________________________________
[gw0] linux -- Python 3.9.11 /usr/bin/python3.9

self = <scipy.io.arff.tests.test_arffread.TestNoData object at 0x3fffa5e0b070>

    def test_nodata(self):
        # The file nodata.arff has no data in the @DATA section.
        # Reading it should result in an array with length 0.
        nodata_filename = os.path.join(data_path, 'nodata.arff')
        data, meta = loadarff(nodata_filename)
        expected_dtype = np.dtype([('sepallength', '<f8'),
                                   ('sepalwidth', '<f8'),
                                   ('petallength', '<f8'),
                                   ('petalwidth', '<f8'),
                                   ('class', 'S15')])
>       assert_equal(data.dtype, expected_dtype)
E       AssertionError: 
E       Items are not equal:
E        ACTUAL: dtype([('sepallength', '>f8'), ('sepalwidth', '>f8'), ('petallength', '>f8'), ('petalwidth', '>f8'), ('class', 'S15')])
E        DESIRED: dtype([('sepallength', '<f8'), ('sepalwidth', '<f8'), ('petallength', '<f8'), ('petalwidth', '<f8'), ('class', 'S15')])

data       = array([],
      dtype=[('sepallength', '>f8'), ('sepalwidth', '>f8'), ('petallength', '>f8'), ('petalwidth', '>f8'), ('class', 'S15')])
expected_dtype = dtype([('sepallength', '<f8'), ('sepalwidth', '<f8'), ('petallength', '<f8'), ('petalwidth', '<f8'), ('class', 'S15')])
meta       = Dataset: iris
        sepallength's type is numeric
        sepalwidth's type is numeric
        petallength's type is numeric
        petalwidth's type is numeric
        class's type is nominal, range is ('Iris-setosa', 'Iris-versicolor', 'Iris-virginica')

nodata_filename = '/var/tmp/portage/dev-python/scipy-1.8.0/work/scipy-1.8.0-python3_9/build/test/lib/scipy/io/arff/tests/data/nodata.arff'
self       = <scipy.io.arff.tests.test_arffread.TestNoData object at 0x3fffa5e0b070>

scipy/io/arff/tests/test_arffread.py:112: AssertionError
```


### SciPy/NumPy/Python version information

1.8.0 1.22.2 sys.version_info(major=3, minor=9, micro=11, releaselevel='final', serial=0)

--- a/scipy/io/arff/tests/test_arffread.py
+++ b/scipy/io/arff/tests/test_arffread.py
@@ -106,7 +106,7 @@ class TestNoData:
         data, meta = loadarff(nodata_filename)
         if sys.byteorder == 'big':
             end = '>'
-        elif sys.byteorder == 'little':
+        else:
             end = '<'
         expected_dtype = np.dtype([('sepallength', f'{end}f8'),
                                    ('sepalwidth', f'{end}f8'),

--- a/scipy/io/arff/tests/test_arffread.py
+++ b/scipy/io/arff/tests/test_arffread.py
@@ -104,10 +104,14 @@ class TestNoData:
         # Reading it should result in an array with length 0.
         nodata_filename = os.path.join(data_path, 'nodata.arff')
         data, meta = loadarff(nodata_filename)
-        expected_dtype = np.dtype([('sepallength', '<f8'),
-                                   ('sepalwidth', '<f8'),
-                                   ('petallength', '<f8'),
-                                   ('petalwidth', '<f8'),
+        if sys.byteorder == 'big':
+            end = '>'
+        elif sys.byteorder == 'little':
+            end = '<'
+        expected_dtype = np.dtype([('sepallength', f'{end}f8'),
+                                   ('sepalwidth', f'{end}f8'),
+                                   ('petallength', f'{end}f8'),
+                                   ('petalwidth', f'{end}f8'),
                                    ('class', 'S15')])
         assert_equal(data.dtype, expected_dtype)
         assert_equal(data.size, 0)

